content/en/docs/usage/kraft_advanced.md

```diff
@@ -0,0 +1,732 @@
+  -D, --dry-run              Perform a dry run.
+  -M, --memory INTEGER       Assign MB memory to the guest.
+  -s, --cpu-sockets INTEGER  Number of guest CPU sockets.
+  -c, --cpu-cores INTEGER    Number of guest cores per socket.
+  -F, --force                Overwrite any existing files in current working
+			     directory.
+  -j, --fast                 Use all CPU cores to build the application.
+  -M, --with-makefile        Create a Unikraft compatible Makefile.
+  -h, --help                 Show this message and exit.
+```
+
+## `kraft init`
+
+You can initialise an empty application, or download an existing one by using `kraft init`.
+You can choose to create a compatible Makefile for the application by using the `-M, --with-makefile` option.
+If you want to use an existing application listed in `kraft list`, you can use the `-t, --template` option.
+
+```console
+$ kraft init -M -t helloworld my-helloworld-app
+[...]
+[INFO    ] Initialized new unikraft application: /home/stefan/projects/unikraft/my-helloworld-app-init
+```
+
+A new directory named `my-helloworld-app/` will be created inside you working directory, and will contain the following file structure:
+
+```text
+.
+|-- CODING_STYLE.md
+|-- Config.uk
+|-- CONTRIBUTING.md
+|-- COPYING.md
+|-- kraft.yaml
+|-- main.c
+|-- MAINTAINERS.md
+|-- Makefile
+|-- Makefile.uk
+|-- monkey.h
+`-- README.md
+```
+
+You can use the generated `Makefile` to invoke the Unikraft build system.
+
+```console
+$ cat Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS :=
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+
+```
+
+You can build your application on top of this basic file structure, by adding source files and modifying the `Makefile.uk` and `Config.uk` files, as presented in the [advanced usage docs](https://unikraft.org/docs/usage/advanced/).
+
+### Overview of the `kraft init` Command
+
+You can get a full list of features by running `kraft init -h` or `kraft init --help`.
+
+```text
+Usage: kraft init [OPTIONS] [NAME]
+
+  Initializes a new unikraft application.
+
+  Start here if this is your first time using (uni)kraft.
+
+Options:
+  -t, --template NAME  Use an existing application as a template.
+  -p, --plat PLAT      Target platform.
+  -m, --arch ARCH      Target architecture.
+  -w, --workdir PATH   Specify an alternative directory for the application
+		       [default is cwd].
+  -M, --with-makefile  Create a Unikraft compatible Makefile.
+  -D, --no-deps        Do not download additional dependencies for application
+		       components.
+  -d, --dump           Dump dependencies into project directory.
+  -F, --force          Overwrite any existing files.
+  -h, --help           Show this message and exit.
+```
+
+## `kraft configure`
+
+You can configure the application by using `kraft configure`.
+The configuration will be saved inside the `.config` file.
```

Comment:
Add some explanation here that `kraft configure` processes the application `kraft.yaml` file and `Config.uk` files in libraries and, based on encountered configuration options, generates the `.config` file.
Point to a documentation link where the `.config` file is explained, as a list of key-value items.
If no documentation, point to a place where the documentation should reside.